oio/api/object_storage.py

```diff
@@ -1182,6 +1182,25 @@ def _m2_chunk_bean(url, meta, content):
                 "pos": meta["pos"],
                 "content": content}
 
+    def object_head(self, account, container, obj, full=False, **kwargs):
+        """
+        Check for the presence of an object in a container.
+
+        :param full: if true, check the presence of the chunks in the rawx
+        :type full: `bool`
+        """
```

Comment:
I propose we extend this API with a `depth` parameter instead of `full`.
- `depth=0`: do not check chunks;
- `depth=1`: check if there are enough chunks to read the object;
- `depth=2`: check if all chunks are present.

Reply:
In the scope of this pull-request, you can stop at `trust_level=0` and `trust_level=2`, letting `trust_level=1` to another pull-request.